addons/block_code/ui/picker/categories/category_factory.gd

```diff
@@ -766,3 +740,29 @@ static func _get_input_blocks() -> Array[Block]:
 				InputMap.action_add_event(action, event)
 
 	return block_list
+
+
+static func get_variable_blocks(variables: Array[VariableResource]):
+	var block_list: Array[Block]
+
+	for variable in variables:
+		var capital_name: String = variable.var_name.capitalize()
+		var camel_name: String = variable.var_name.to_snake_case()
+		var type_string: String = Types.VARIANT_TYPE_TO_STRING[variable.var_type]
+
+		var b = BLOCKS["parameter_block"].instantiate()
+		b.variant_type = variable.var_type
+		b.block_format = capital_name
+		b.statement = camel_name
```

Comment:
It's a bit strange that the entered, displayed and formatted variables are all different. Why not just use the variable name as was entered by the user? What happens when the user creates `foo` and `Foo` variables? Legally these are two different variables, but you won't be able to distinguish them in the UI and they'll conflict in the generated script.

Reply:
Yea my idea was to make them look pretty but it's probably better to just have the display and formatted be the same as what the user types. Will also have to check for invalid/duplicate names.